### PR TITLE
Refactor XIBC

### PR DESCRIPTION
**Pull request by Kukoomomo**
Close: #157 

- [x] Change proto 
- [x] Add crossChain
- [x] Packet evm_hook
- [x] Add tests

### CODE REVIEW

File: proto/xibc/core/packet/v1/packet.proto [hunk 1]
```diff
@@ -9,21 +9,41 @@ import "gogoproto/gogo.proto";
...
-  // with an earlier sequence number must be sent and received before a Packet
-  // with a later sequence number.
-  uint64 sequence = 1;
   // identifies the chain id of the sending chain.
-  string source_chain = 2 [ (gogoproto.moretags) = "yaml:\"source_chain\"" ];
+  string source_chain = 1 [ (gogoproto.moretags) = "yaml:\"source_chain\"" ];
   // identifies the chain id of the receiving chain.
-  string destination_chain = 3
+  string destination_chain = 2
       [ (gogoproto.moretags) = "yaml:\"destination_port\"" ];
```
Comment (Zer0ne001) [inline proto/xibc/core/packet/v1/packet.proto:16]: fix typo `destination_port`

Comment (Zer0ne001) [reply proto/xibc/core/packet/v1/packet.proto:16]: to destination_chain, the `port` concept has been removed

File: x/xibc/core/packet/keeper/packet.go [hunk 2]
```diff
@@ -122,42 +123,10 @@ func (k Keeper) RecvPacket(ctx sdk.Context, msg *types.MsgRecvPacket) error {
...
-			SrcChain:   packet.GetSourceChain(),
-			DstChain:   packet.GetDestChain(),
-			RelayChain: packet.GetRelayChain(),
-			Ports:      packet.GetPorts(),
-			DataList:   packet.GetDataList(),
+			Packet: msg.Packet,
 		},
 	)
 
-	if packet.GetRelayChain() == chainName {
```
Comment (Zer0ne001) [inline x/xibc/core/packet/keeper/packet.go:134]: if (dest_chain != chain_name && dest_client_exist) {
    handle relay chain logic
}

File: x/xibc/core/packet/keeper/packet.go [hunk 3]
```diff
@@ -301,70 +268,10 @@ func (k Keeper) AcknowledgePacket(
...
-			RelayChain: packet.GetRelayChain(),
-			Ports:      packet.GetPorts(),
-			DataList:   packet.GetDataList(),
-			Ack:        msg.Acknowledgement,
+			Packet: msg.Packet,
+			Ack:    msg.Acknowledgement,
 		},
 	)
 
-	if packet.GetRelayChain() == chainName {
```
Comment (Zer0ne001) [inline x/xibc/core/packet/keeper/packet.go:314]: if (src_chain != chain_name) {
    handle relay chain logic
}

File: x/xibc/core/packet/keeper/packet.go (hunk 2 above)
Comment (Kukoomomo) [reply x/xibc/core/packet/keeper/packet.go:134]: If we have the work for relay packet by teleport to be the relayChain . relayer may only relay the packet with the channel(srcChain -> destChain) , how do relayer know that the teleport is relayChain and the teleport not the srcChain and destChain?

File: proto/xibc/core/packet/v1/event.proto [hunk 4]
```diff
@@ -8,42 +8,22 @@ option go_package = "github.com/teleport-network/teleport/x/xibc/core/packet/typ
 
 // EventSendPacket is emitted on send packet
 message EventSendPacket {
-  string sequence = 1;
-  string src_chain = 2;
-  string dst_chain = 3;
-  string relay_chain = 4;
-  repeated string ports = 5;
-  repeated bytes data_list = 6;
```
[the remark this answers is not in the document]
Comment (Kukoomomo) [reply proto/xibc/core/packet/v1/event.proto:16]: Add sequence src_chain dst_chain for this packet , Is this method ok?

**Review by Zer0ne001 — APPROVED** [review]

**Review by KaelTop — APPROVED** [review]

**Pull request merged**